### ISSUE TITLE
[🐛 bug report] In the function Adam.step() of autodiff.py, after writng to the dictionary of params, there is no calling of update()

### ISSUE THREAD

**Issue by JiahaoGeng** [mitsuba-renderer/mitsuba2#291]
## Summary
Refer to the subscribtion of "update()" in Python API Reference, this function should be called at the end of a sequence of writes to the dictionary. But in Adam.step(), after self.params[k] = v，there is no calling of update(). The code is as bellow:

#################autodiff.py, class Adam######################
def step(self):
        """ Take a gradient step """
        self.t += 1

        from mitsuba.core import Float
        lr_t = ek.detach(Float(self.lr * ek.sqrt(1 - self.beta_2**self.t) /
                               (1 - self.beta_1**self.t), literal=False))

        for k, p in self.params.items():
            g_p = ek.gradient(p)
            size = ek.slices(g_p)

            if size == 0:
                continue
            elif size != ek.slices(self.state[k][0]):
                # Reset state if data size has changed
                self._reset(k)

            m_tp, v_tp = self.state[k]
            m_t = self.beta_1 * m_tp + (1 - self.beta_1) * g_p
            v_t = self.beta_2 * v_tp + (1 - self.beta_2) * ek.sqr(g_p)
            self.state[k] = (m_t, v_t)

            u = ek.detach(p) - lr_t * m_t / (ek.sqrt(v_t) + self.epsilon)
            u = type(p)(u)
            ek.set_requires_gradient(u)
            self.params[k] = u
#################################################

**Comment by Speierers** [comment]
Hi @JiahaoGeng ,

This is a known issue. Let me quickly fix this on `master`.
